packages/camera/camera_web/example/integration_test/camera_test.dart

```diff
@@ -297,5 +298,170 @@ void main() {
+
+          await eventExpectation;
+          expect('inactive', camera.mediaRecorder!.state);*/
+      });
+
+      testWidgets(
+          'throws PlatformException '
+          'when maxVideoDuration is 0 milliseconds or less', (tester) async {
+        final camera = Camera(
+          textureId: 1,
+          cameraSettings: cameraSettings,
+        );
+
+        await camera.initialize();
+        await camera.play();
+        expect(
+            () => camera.startVideoRecording(maxVideoDuration: Duration.zero),
+            throwsA(predicate<PlatformException>(
+                (ex) => ex.code == CameraErrorCode.notSupported.toString())));
+      });
+    });
+
+    group('pauseVideoRecording', () {
+      testWidgets('pauses a video recording', (tester) async {
+        final camera = Camera(
+          textureId: 1,
+          cameraSettings: cameraSettings,
+        );
+
+        await camera.initialize();
+        await camera.play();
+        await camera.startVideoRecording();
+
+        await camera.pauseVideoRecording();
+
+        expect('paused', camera.mediaRecorder!.state);
+      });
+
+      testWidgets(
+          'throws a PlatformException '
+          'if no recording was started', (tester) async {
+        final camera = Camera(
+          textureId: 1,
+          cameraSettings: cameraSettings,
+        );
+
+        await camera.initialize();
+        await camera.play();
+
+        expect(
+            camera.pauseVideoRecording,
+            throwsA(predicate<PlatformException>((ex) =>
+                ex.code ==
+                CameraErrorCode.mediaRecordingNotStarted.toString())));
+      });
+    });
+
+    group('resumeVideoRecording', () {
+      testWidgets('resumes a video recording', (tester) async {
+        final camera = Camera(
+          textureId: 1,
+          cameraSettings: cameraSettings,
+        );
+
+        await camera.initialize();
+        await camera.play();
+
+        await camera.startVideoRecording();
+
+        await camera.pauseVideoRecording();
+
+        await camera.resumeVideoRecording();
+
+        expect('recording', camera.mediaRecorder!.state);
+      });
+
+      testWidgets(
+          'throws a PlatformException '
+          'if no recording was started', (tester) async {
+        final camera = Camera(
+          textureId: 1,
+          cameraSettings: cameraSettings,
+        );
+
+        await camera.initialize();
+        await camera.play();
+
+        expect(
+            camera.resumeVideoRecording,
+            throwsA(predicate<PlatformException>((ex) =>
+                ex.code ==
+                CameraErrorCode.mediaRecordingNotStarted.toString())));
+      });
+    });
+
+    group('stopVideoRecording', () {
+      testWidgets('stops a video recording and returns a File', (tester) async {
+        // TODO(abausg) There seems to be an issue with removing the listener and stopping the recorder
```

Comment:
Are you still tracking this down @ABausG?

Reply:
I've managed to get this test to work 🙌